lib/galaxy/webapps/galaxy/api/history_contents.py

```diff
@@ -431,16 +431,26 @@ def index_jobs_summary(
         summary="Download the content of a dataset collection as a `zip` archive.",
         response_class=StreamingResponse,
     )
+    def download_history_dataset_collection(
+        self,
+        trans: ProvidesHistoryContext = DependsOnTrans,
+        history_id: EncodedDatabaseIdField = HistoryIDPathParam,
+        id: EncodedDatabaseIdField = HistoryHDCAIDPathParam,
+    ):
+        """Download the content of a history dataset collection as a `zip` archive
+        while maintaining approximate collection structure.
+        """
+        archive = self.service.get_dataset_collection_archive_for_download(trans, id)
+        return StreamingResponse(archive.get_iterator(), headers=archive.get_headers())
+
     @router.get(
-        "/api/dataset_collection/{id}/download",
+        "/api/dataset_collections/{id}/download",
         summary="Download the content of a dataset collection as a `zip` archive.",
         response_class=StreamingResponse,
-        tags=["dataset collections"],
     )
     def download_dataset_collection(
         self,
         trans: ProvidesHistoryContext = DependsOnTrans,
-        history_id: EncodedDatabaseIdField = HistoryIDPathParam,
```

Comment:
This is what was causing the validation error, the path parameter is required. The way of doing it optional is making it an Optional `Query` parameter, then it will be required in the other route alias as it is part of the path. I just did something similar in #13681